### PR TITLE
feature flag dependencies

### PR DESCRIPTION
**Pull request by snopoke**
## Product Description
In some cases feature flags depend on each other. Although this is not ideal since it adds complexity we already have a number of dependencies. This PR makes those dependencies explicit and simplifies the process for enabling flags that depend on others.

image

## Technical Summary
* Dependencies are statically declared in the code
* At least one namespace must be shared between the dependent toggle and its dependencies

### Safety story
Changes to the toggles would prevent the Django process from starting if there were errors so would be caught by the build. The view changes have been tested locally.

### Automated test coverage
None

### QA Plan
None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/toggles/__init__.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ class StaticToggle(object):
     def __init__(self, slug, label, tag, namespaces=None, help_link=None,
                  description=None, save_fn=None, enabled_for_new_domains_after=None,
                  enabled_for_new_users_after=None, relevant_environments=None,
-                 notification_emails=None):
+                 notification_emails=None, dependent_toggles=None):
```
Comment (esoergel) [inline corehq/toggles/__init__.py:126]: nit: maybe `dependent_on_toggles`, `dependency_toggles`, or `parent_toggles` to make the directionality clearer?

File: corehq/apps/toggle_ui/views.py [hunk 2]
```diff
@@ -357,16 +375,7 @@ def set_toggle(request, toggle_slug):
...
-    if static_toggle.set(item=item, enabled=enabled, namespace=namespace):
-        action = ToggleAudit.ACTION_ADD if enabled else ToggleAudit.ACTION_REMOVE
-        ToggleAudit.objects.log_toggle_action(
-            toggle_slug, request.user.username, [namespaced_item(item, namespace)], action
-        )
-
-    if enabled:
-        _notify_on_change(static_toggle, [item], request.user.username)
-
-    _clear_cache_for_toggle(namespace, item)
```
Comment (esoergel) [inline corehq/apps/toggle_ui/views.py:369]: I notice that the indent level for these last two blocks changed when you moved them, but I think the new behavior is more correct, so I'm guessing that was intentional.

**Review by esoergel — APPROVED** [review]
This is cool - it could be used to simplify the case search index stuff too, though that's obviously out of scope.

File: corehq/apps/toggle_ui/views.py (hunk 2 above)
Comment (snopoke) [reply corehq/apps/toggle_ui/views.py:369]: Yes, these are only needed if the toggle was actually updated.

**Review by kaapstorm — APPROVED** [review]
This is great!

I like the suggestion of making the directionality more obvious.

File: corehq/toggles/__init__.py (hunk 1 above)
Comment (snopoke) [reply corehq/toggles/__init__.py]: 91f7ed32b8f

**Pull request merged**